### PR TITLE
feat(localization): Support for Spanish

### PR DESCRIPTION
**Pull request by sachaarbonel**
- Support for Spanish
- Fix some typos in French

### CODE REVIEW

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 1]
```diff
@@ -0,0 +1,364 @@
+part of 'stream_chat_localizations.dart';
+
+/// The translations for Spanish (`es`).
+class StreamChatLocalizationsEs extends GlobalStreamChatLocalizations {
+  /// Create an instance of the translation bundle for Spanish.
+  const StreamChatLocalizationsEs({String localeName = 'es'})
+      : super(localeName: localeName);
+
+  @override
+  String get launchUrlError => 'No se puede lanzar la url';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:10]:
```suggestion
  String get launchUrlError => 'No se pudo abrir la url';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+
+  @override
+  String get loadingUsersError => 'Error de carga del usuario';
+
+  @override
+  String get noUsersLabel => 'No hay usuarios actualmente';
+
+  @override
+  String get retryLabel => 'Inténtalo de nuevo';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:19]:
```suggestion
  String get retryLabel => 'Inténtelo de nuevo';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+
+  @override
+  String get threadReplyLabel => 'Responder al hilo de discusión';
+
+  @override
+  String get onlyVisibleToYouText => 'Sólo visible para usted';
+
+  @override
+  String threadReplyCountText(int count) =>
+      '$count Respuestas al hilo de discusión';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:45]:
```suggestion
      '$count respuestas al hilo de discusión';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+
+  @override
+  String get messageReactionsLabel => 'Reacciones a los mensajes';
+
+  @override
+  String get emptyChatMessagesText => 'Todavía no hay charlas aquí...';
+
+  @override
+  String threadSeparatorText(int replyCount) {
+    if (replyCount == 1) return '1 Respuesta';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:90]:
```suggestion
    if (replyCount == 1) return '1 respuesta';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+    return '$replyCount Respuestas';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:91]:
```suggestion
    return '$replyCount respuestas';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+  }
+
+  @override
+  String get emptyMessagesText => 'Actualmente no hay mensajes';
+
+  @override
+  String get genericErrorText => 'Hubo un problema';
+
+  @override
+  String get loadingMessagesError => 'Mensajes de error de carga';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:71]:
```suggestion
  String get loadingMessagesError => 'Hubo un error mientras se cargaba el mensaje';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+  String get addACommentOrSendLabel => 'Añadir un comentario o enviar';
+
+  @override
+  String get searchGifLabel => 'Búsqueda de GIFs';
+
+  @override
+  String get writeAMessageLabel => 'Escribir un mensaje';
+
+  @override
+  String get instantCommandsLabel => 'Mandos instantáneos';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:117]:
```suggestion
  String get instantCommandsLabel => 'Comandos instantáneos';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 8]
```diff
@@ -0,0 +1,364 @@
...
+  @override
+  String get copyMessageLabel => 'Copiar el mensaje';
+
+  @override
+  String get editMessageLabel => 'Editar el mensaje';
+
+  @override
+  String toggleResendOrResendEditedMessage({required bool isUpdateFailed}) {
+    if (isUpdateFailed) return 'Reenviar el mensaje modificado';
+    return 'Enviar de vuelta';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:228]:
```suggestion
    return 'Reenviar';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 9]
```diff
@@ -0,0 +1,364 @@
...
+    } else if (date == yesterday) {
+      return 'ayer';
+    } else {
+      return 'el ${Jiffy(date).MMMd}';
+    }
+  }
+
+  @override
+  String sentAtText({required DateTime date, required DateTime time}) =>
+      'Enviado ${_getDay(date)} a ${Jiffy(time.toLocal()).format('HH:mm')}';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:252]:
```suggestion
      'Enviado el ${_getDay(date)} a las ${Jiffy(time.toLocal()).format('HH:mm')}';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 10]
```diff
@@ -0,0 +1,364 @@
...
+
+  @override
+  String get todayLabel => 'Hoy';
+
+  @override
+  String get yesterdayLabel => 'Ayer';
+
+  @override
+  String get channelIsMutedText => 'El canal está cortado';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:261]:
```suggestion
  String get channelIsMutedText => 'El canal está silenciado';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 11]
```diff
@@ -0,0 +1,364 @@
...
+
+  @override
+  String get letsStartChattingLabel => '¡Empecemos a charlar!';
+
+  @override
+  String get sendingFirstMessageLabel =>
+      '¿Qué le parece enviar su primer mensaje a un amigo?';
+
+  @override
+  String get startAChatLabel => 'Iniciar una discusión';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:274]:
```suggestion
  String get startAChatLabel => 'Iniciar una conversación';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 12]
```diff
@@ -0,0 +1,364 @@
...
+
+  @override
+  String get deleteConversationQuestion =>
+      '¿Estás seguro de que quieres borrar esta conversación?';
+
+  @override
+  String get streamChatLabel => 'Stream Chat';
+
+  @override
+  String get searchingForNetworkText => 'Búsqueda en la red';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:290]:
```suggestion
  String get searchingForNetworkText => 'Buscando red';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 13]
```diff
@@ -0,0 +1,364 @@
...
+
+  @override
+  String get offlineLabel => 'Sin conexión...';
+
+  @override
+  String get tryAgainLabel => 'Inténtalo de nuevo';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:296]:
```suggestion
  String get tryAgainLabel => 'Inténtelo de nuevo';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 14]
```diff
@@ -0,0 +1,364 @@
...
+
+  @override
+  String membersCountText(int count) {
+    if (count == 1) return '1 Membre';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:300]:
```suggestion
    if (count == 1) return '1 miembro';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 15]
```diff
@@ -0,0 +1,364 @@
...
+    return '$count Membres';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:301]:
```suggestion
    return '$count miembros';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 16]
```diff
@@ -0,0 +1,364 @@
...
+  String watchersCountText(int count) {
+    if (count == 1) return '1 En línea';
+    return '$count En línea';
+  }
+
+  @override
+  String get viewInfoLabel => 'Ver información';
+
+  @override
+  String get leaveGroupLabel => 'Dejar el Grupo';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:314]:
```suggestion
  String get leaveGroupLabel => 'Salir del Grupo';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 17]
```diff
@@ -0,0 +1,364 @@
...
+
+  @override
+  String get leaveLabel => 'DEJAR';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:317]:
```suggestion
  String get leaveLabel => 'SALIR';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 18]
```diff
@@ -0,0 +1,364 @@
...
+
+  @override
+  String get leaveConversationLabel => 'Dejar la conversación';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:320]:
```suggestion
  String get leaveConversationLabel => 'Salir de la conversación';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart [hunk 19]
```diff
@@ -0,0 +1,364 @@
...
+
+  @override
+  String get leaveConversationQuestion =>
+      '¿Estás seguro de que quieres dejar esta conversación?';
```
Comment (JcMinarro) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_es.dart:324]:
```suggestion
      '¿Estás seguro de que quiere salir de esta conversación?';
```

**Review by JcMinarro — CHANGES_REQUESTED** [review]

**Comment by xsahil03x** [comment]
Nice try @sachaarbonel

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations.dart [hunk 20]
```diff
@@ -19,12 +21,7 @@ part 'stream_chat_localizations_hi.dart';
 /// See also:
 ///
 ///  * [getStreamChatTranslation], whose documentation describes these values.
-const kStreamChatSupportedLanguages = {
-  'en',
-  'hi',
-  'fr',
-  'it',
-};
+const kStreamChatSupportedLanguages = {'en', 'hi', 'fr', 'it', 'es'};
```
Comment (imtoori) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations.dart:24]: missing column

**Review by imtoori — CHANGES_REQUESTED** [review]
this is great
just a couple of things: 
- missing column
- can you add the Spanish support to the package readme and the guide on docusaurus as well?

**Review by JcMinarro — DISMISSED** [review]
Spanish translations are good now ;)

**Comment by xsahil03x** [comment]
> Spanish translations are good now ;)

Thanks @JcMinarro for a great review 😄

**Review by JcMinarro — APPROVED** [review]

**Review by xsahil03x — APPROVED** [review]
LGTM

**Pull request merged**